### ISSUE TITLE
[a11y]: SidePanel: use <aside> rather than role="complementary"

### ISSUE THREAD

**Issue by wkeese** [carbon-design-system/ibm-products#6314]
### Package

@carbon/ibm-products

### Browser

Chrome

### Operating System

MacOS

### Package version

2.52.0

### React version

18

### Automated testing tool and ruleset

SonarQube

### Description

SonarQube has started to give warnings to [use elements in preference to attributes](https://sonarsource.github.io/rspec/#/rspec/S6819), and additionally, MDN's documentation for role="complementary" says [to prefer `<aside>` over setting the complementary role](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/complementary_role).

Thus, I suggest to change `SidePanel` to use an `<aside>` rather than a `<div role=complementary>`.

### WCAG 2.1 Violation

Unknown

### Reproduction/example

https://carbon-for-ibm-products.netlify.app/?path=/story/ibm-products-components-side-panel-sidepanel--slide-over&globals=viewport:basic

### Steps to reproduce

N/A

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/ibm-products/blob/main/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/ibm-products/issues) for duplicate problems

**Comment by amal-k-joy** [comment]
recommended as [here](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/complementary_role)

**Comment by devadula-nandan** [comment]
include for web components too

**Comment by elycheea** [comment]
`<aside>`/`role="complementary"` may be more appropriate for slide-in modals rather than slide-over modals which would be more of `dialog`s.

For future iterations of side panel, we could potentially consider using the `as` prop[^1] for aside/dialog instead of slide-in or slide-over since how these behave is actually more closely related to their actual roles instead.

[^1]: https://react.carbondesignsystem.com/?path=/docs/components-link--overview
